pkg/apis/cartographer/v1alpha1/workload_source_printer.go

```diff
@@ -0,0 +1,190 @@
+/*
+Copyright 2021 VMware, Inc.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"io"
+	"strings"
+	"time"
+
+	metav1beta1 "k8s.io/apimachinery/pkg/apis/meta/v1beta1"
+
+	"github.com/vmware-tanzu/apps-cli-plugin/pkg/cli-runtime/printer"
+	"github.com/vmware-tanzu/apps-cli-plugin/pkg/cli-runtime/printer/table"
+)
+
+type resourcesQuery struct {
+	resourceName  string
```

Comment:
Looking for values based on the resource name is *very* fragile. The resource name is a reference that only has meaning within the context of a specific supply chain. There is no requirement that any supply chain have a resource of a specific name.

As implemented, callers to this method will only return meaningful results for the OOTB supply chains that ship with TAP. Even then, since it's an internal reference, there's no promise those values won't change in a future TAP release.